tests/test_configurations/test_base.py

```diff
@@ -206,7 +206,10 @@ class TestAgentConfig:
     def test_from_json_and_to_json(self, agent_path):
         """Test the 'from_json' method and 'to_json' work correctly."""
         f = open(agent_path)
-        original_json = yaml.safe_load(f)
+        original_jsons = list(yaml.safe_load_all(f))
+        components = original_jsons[1:]
+        original_json = original_jsons[0]
+        original_json["component_configurations"] = components
```

Comment:
This does not look like a good solution; we should not require changes here..

Reply:
what am I missing?

Reply:
`safe_load_all` returns a list and it is the only reasonable way to load a multi-paged YAML file with the `YAML` APIs.

Reply:
Also, the `AgentConfig.json` doesn't return a list, but a dictionary. 
Please read this comment: https://github.com/fetchai/agents-aea/pull/1674/files#r472072161